frontend/src/js/components/SeatSelector.jsx

```diff
@@ -19,18 +33,43 @@ var SeatSelector = React.createClass({
     }
 
     // different price groups among seats in this show; prices[0] is the most expensive one
-    var getBasePrice = function(seat) {
-      return seat.prices[0].price;
+    var getBasePrice = function(section) {
+      return section.discount_groups[0].price;
     };
-    var prices = _.uniq(this.props.seats.map(getBasePrice)).sort().reverse();
+    var prices = _.uniq(_.values(this.props.show.get('sections')).map(getBasePrice)).sort().reverse();
+
+    var statuses = this.getSeatStatuses(this.props.conflictingSeatIds, this.props.chosenSeatIds, this.props.reservedSeatIds);
 
     return (
       <div className={divClass}>
         <h2>Paikkojen valinta <small>2/5</small></h2>
         <div className='theaterLayout' style={{backgroundImage: 'url(public/img/venues/venue_1.png)'}}>
-          {this.props.seats.map(function(seat) {
-            var priceGroup = prices.indexOf(getBasePrice(seat));
-            return <Seat seat={seat} priceGroup={priceGroup} status={seat.status} key={seat.id} onClick={this.props.active ? this.props.onSeatClicked.bind(null, seat) : null} />;
+          {_.values(this.props.venue.get('sections')).map(function(section) {
+            var showSection = this.props.show.get('sections')[section.id];
+            if (!showSection) {
+              return null; // if section is set to active=0 in DB table 'nk2_prices'
+            }
+            var price = _.pluck(showSection.discount_groups, 'price');
+            var priceGroup = prices.indexOf(price[0]);
```

Comment:
`price[0]` is same as `showSection.discount_groups[0].price`. But the `price`-array contains all prices for the given section (normal and discounted), and priceGroup is there to set the color of the seats